azurerm/internal/services/datafactory/tests/resource_arm_data_factory_linked_service_blob_storage_test.go

```diff
@@ -0,0 +1,226 @@
+				),
+			},
+			{
+				Config: testAccAzureRMDataFactoryLinkedServiceBlobStorage_update2(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataFactoryLinkedServiceBlobStorageExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "parameters.%", "3"),
+					resource.TestCheckResourceAttr(data.ResourceName, "annotations.#", "2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "additional_properties.%", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "description", "test description 2"),
+				),
+			},
+			data.ImportStep("connection_string"),
+		},
+	})
+}
+
+func testCheckAzureRMDataFactoryLinkedServiceBlobStorageExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).DataFactory.LinkedServiceClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		dataFactoryName := rs.Primary.Attributes["data_factory_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Data Factory: %s", name)
+		}
+
+		resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			return fmt.Errorf("Bad: Get on dataFactoryLinkedServiceClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Data Factory Linked Service BlobStorage %q (data factory name: %q / resource group: %q) does not exist", name, dataFactoryName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMDataFactoryLinkedServiceBlobStorageDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DataFactory.LinkedServiceClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_data_factory_linked_service_blob_storage" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		dataFactoryName := rs.Primary.Attributes["data_factory_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Data Factory Linked Service BlobStorage still exists:\n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMDataFactoryLinkedServiceBlobStorage_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```

Comment:
could we make the resource group for all tests
```suggestion
  name     = "acctestRG-df-%d"
```